lib/elixir/lib/string.ex

```diff
@@ -1307,7 +1300,7 @@ defmodule String do
       ["abc\0"]
 
       iex> String.chunk(<<?a, ?b, ?c, 0, 0x0FFFF::utf8>>, :valid)
-      ["abc\0", <<0x0FFFF::utf8>>]
+      ["abc\0" <> <<0x0FFFF::utf8>>]
```

Comment:
We still need an example here. Maybe we should change `0x0FFFF::utf8` to `0xFFFF::utf16`?